knowledge_verificator/utils/configuration_parser.py

```diff
@@ -0,0 +1,131 @@
+"""Module with the parser for YAML configuration files."""
+
+from dataclasses import dataclass
+from enum import Enum
+from pathlib import Path
+from typing import Any
+import yaml  # type: ignore[import-untyped]
+
+
+class OperatingMode(Enum):
+    """
+    Available mode of operating for the application.
+
+    These modes mean:
+    - EXPERIMENT - implemented experiments on available language models
+        are run. These experiments aim at measuring quality of results
+        produced by the models.
+    - CLI - interactive command-line interface mode.
+        It allows a user to perform actions in terminal.
+    - BACKEND - an HTTP server is started, which serves as backend.
+        Graphical user interface (GUI) in a form of webpage is served
+        as frontend.
+    """
+
+    EXPERIMENT = 'EXPERIMENT'
+    CLI = 'CLI'
+    BACKEND = 'BACKEND'
+
+
+@dataclass
+class Configuration:
+    """
+    A dataclass storing a parsed configuration.
+
+    Names of the attributes are used in this database are used in
+    a YAML configuration file.
+
+    Attributes:
+        learning_materials (Path): Path to a directory containing learning materials.
+        logging_level (str): One of `DEBUG`, `INFO`, `WARNING`, `ERROR`,
+            `CRITICAL`. Minimal level of a message to be logged.
+        mode (OperatingMode): Operating mode of application. Different mode serve
+            different purposes.
+        experiment_implementation (Path): Path to a directory containing
+            implementation of experiments on language models.
+        experiment_results (Path): Path to a directory, where results
+            should be saved.
+    """
+
+    learning_materials: Path
+    logging_level: str
+    mode: OperatingMode
+    experiment_implementation: Path
+    experiment_results: Path
+    production_mode: bool
+
+    def __init__(
+        self,
+        **kwargs,  # Records received from a YAML file.
+    ) -> None:
+        # Assign values to attributes.
+        for attribute, value in kwargs.items():
+            self.__setattr__(attribute, value)
+
+        # Convert to a proper datatypes.
+        self.learning_materials: Path = Path(kwargs['learning_materials'])
+        self.mode: OperatingMode = OperatingMode(kwargs['mode'])
+        self.experiment_implementation = Path(
+            kwargs['experiment_implementation']
+        )
+        self.experiment_results = Path(kwargs['experiment_results'])
+
+
+class ConfigurationParser:
+    """Class, which loads and parses a YAML configuration."""
+
+    def __init__(self, configuration_file: Path | str) -> None:
+        """Load a YAML configuration file."""
+        self._config_file: Path | None = None
+        self._config_data: dict = {}
+        self.load_configuration(configuration_path=configuration_file)
```

Comment:
There is only one argument there... why would you use `configuration_path=` when even the parameter has the exact same name?

Reply:
More parameters may come later. If the order of arguments changes, there will be no need to modify this line.

Reply:
Okay, there will not be any further argument. I am removing  `configuration_path`.